### PR TITLE
[2847] Add toggle component

### PR DESCRIPTION
**Pull request by RTAndrew**
image

### CODE REVIEW

**Review by RTAndrew — COMMENTED** [review]
Check comments

**Comment by henriquemacedo** [comment]
Typo on the component folder name:
`toogle` → `toggle`

File: src/atoms/toogle/stories/toogle.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      control: {
+        type: 'boolean',
+      },
+    },
+    checked: {
+      control: {
+        type: 'boolean',
+      },
+    },
+  },
```
Comment (henriquemacedo) [inline src/atoms/toogle/stories/toogle.stories.tsx:24]:
```suggestion
```

You can remove `argTypes` in this case. This is useful when we have multiple stories for the same component, and we want to create default arguments for all.

File: src/atoms/toogle/index.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  const handleClick = (value: boolean) => {
+    if (disabled) return;
+
+    setState(value);
+    return onClick(value);
+  };
+
+  return (
+    <Styles.Switcher
```
Comment (henriquemacedo) [inline src/atoms/toogle/index.tsx:25]:
```suggestion
    <Styles.Switcher
      className={className}
```

Add `className` to make it easier for anyone who wants to customize some element of the toggle.

File: src/atoms/toogle/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+
+import * as Styles from './styles';
+import { ToggleProps } from './types';
+
+const Toggle = ({
```
Comment (henriquemacedo) [inline src/atoms/toogle/index.tsx:6]:
```suggestion
const Toggle = ({
  className = 'toggle',
```

Add "toggle" as `className` default.

File: src/atoms/toogle/styles.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    label {
+      cursor: pointer;
+    }
+
+    .wrapper {
+      margin: 0 ${rem('10px')};
+      display: inline-block;
+      vertical-align: middle;
+      width: ${rem('80px')};
+      height: ${rem('40px')};
```
Comment (henriquemacedo) [inline src/atoms/toogle/styles.tsx:22]:
```suggestion
      width: ${rem('48px')};
      height: ${rem('24px')};
```

File: src/atoms/toogle/styles.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  padding: 0;
+
+  > div {
+    label {
+      cursor: pointer;
+    }
+
+    .wrapper {
+      margin: 0 ${rem('10px')};
+      display: inline-block;
```
Comment (henriquemacedo) [inline src/atoms/toogle/styles.tsx:19]:
```suggestion
    label {
      cursor: pointer;

      &:first-child {
        margin-right: ${rem('10px')};
      }

      &:last-child {
        margin-left: ${rem('10px')};
      }
    }

    .wrapper {
      display: inline-block;
```

This way the toggle doesn't have the margins even when it doesn't have the labels.

File: src/atoms/toogle/index.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  isLabelVisible = false,
+  disabled = false,
+  checked = true,
```
Comment (henriquemacedo) [inline src/atoms/toogle/index.tsx:9]:
```suggestion
  checked = false,
```

I think `false` is a better value for the default state. What do you think?

**Comment by henriquemacedo** [comment]
We can add something like an `onClick` on the `.switcher` component to change the state when we click on the blank circle. What do you think?

File: src/atoms/toogle/index.tsx (hunk 2 above)
Comment (RTAndrew) [reply src/atoms/toogle/index.tsx:25]: I will extend the `HTMLDivElement` attributes and give the developer much more flexibility when using the component.

**Pull request merged**